From e1d99b9b34af90af5610ada0c51ec626cea352e5 Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Fri, 21 Jan 2022 04:53:50 -0800
Subject: [PATCH 1/3] Sync
 0001-Extend-struct-r_debug-to-support-multiple-namespaces.patch

Sync with glibc-2.34-20.fc35
---
 ...debug-to-support-multiple-namespaces.patch | 35 +++++++++----------
 1 file changed, 16 insertions(+), 19 deletions(-)

diff --git a/0001-Extend-struct-r_debug-to-support-multiple-namespaces.patch b/0001-Extend-struct-r_debug-to-support-multiple-namespaces.patch
index 3f29dc0..a6dd7c4 100644
--- a/0001-Extend-struct-r_debug-to-support-multiple-namespaces.patch
+++ b/0001-Extend-struct-r_debug-to-support-multiple-namespaces.patch
@@ -85,25 +85,22 @@ diff --git a/elf/Makefile b/elf/Makefile
 index d05f410592..9dd45d2001 100644
 --- a/elf/Makefile
 +++ b/elf/Makefile
-@@ -35,7 +35,8 @@ dl-routines	= $(addprefix dl-,load lookup object reloc deps \
- 				  execstack open close trampoline \
- 				  exception sort-maps lookup-direct \
- 				  call-libc-early-init write \
--				  thread_gscope_wait tls_init_tp)
-+				  thread_gscope_wait tls_init_tp \
-+				  debug-symbols)
- ifeq (yes,$(use-ldconfig))
- dl-routines += dl-cache
- endif
-@@ -203,7 +204,7 @@ tests += restest1 preloadtest loadfail multiload origtest resolvfail \
- 	 tst-tls16 tst-tls17 tst-tls18 tst-tls19 tst-tls-dlinfo \
- 	 tst-align tst-align2 \
- 	 tst-dlmodcount tst-dlopenrpath tst-deep1 \
--	 tst-dlmopen1 tst-dlmopen3 \
-+	 tst-dlmopen1 tst-dlmopen3 tst-dlmopen4 \
- 	 unload3 unload4 unload5 unload6 unload7 unload8 tst-global1 order2 \
- 	 tst-audit1 tst-audit2 tst-audit8 tst-audit9 \
- 	 tst-addr1 tst-thrlock \
+@@ -52,6 +52,7 @@ dl-routines = \
+   dl-call-libc-early-init \
+   dl-close \
+   dl-debug \
++  dl-debug-symbols \
+   dl-deps \
+   dl-exception \
+   dl-execstack \
+@@ -371,6 +372,7 @@ tests += \
+   tst-dlmodcount \
+   tst-dlmopen1 \
+   tst-dlmopen3 \
++  tst-dlmopen4 \
+   tst-dlmopen-dlerror \
+   tst-dlmopen-gethostbyname \
+   tst-dlopenfail \
 @@ -1242,6 +1243,8 @@ $(objpfx)tst-dlmopen2.out: $(objpfx)tst-dlmopen1mod.so
  
  $(objpfx)tst-dlmopen3.out: $(objpfx)tst-dlmopen1mod.so
-- 
2.34.1

